src/highdicom/sr/templates/tid3700.py

```diff
@@ -0,0 +1,1213 @@
+            if not isinstance(prior_report_for_current_patient,
+                              CompositeContentItem):
+                raise TypeError(
+                    'Argument "prior_report_for_current_patient" must ' +
+                    'have type CompositeContentItem.'
+                )
+            content.append(prior_report_for_current_patient)
+        if source_of_measurement is not None:
+            if not isinstance(source_of_measurement, WaveformContentItem):
+                raise TypeError(
+                    'Argument "source_of_measurement" must have type ' +
+                    'WaveformContentItem.'
+                )
+            content.append(source_of_measurement)
+        if len(content) > 0:
+            item.ContentSequence = content
+        super().__init__([item])
+
+
+class ECGFinding(Template):
+    """:sct:`EV 271921002`
+    ECG Finding
+    """
+
+    def __init__(
+        self,
+        value: Union[CodedConcept, Code],
+        equivalent_meaning_of_value: Optional[str] = None,
+        ecg_findings: Optional[Sequence["ECGFinding"]] = None
+    ) -> None:
+        item = CodeContentItem(
+            name=CodedConcept(
+                value='271921002',
+                meaning='ECG Finding',
+                scheme_designator='SCT'
+            ),
+            value=value,
+            relationship_type=RelationshipTypeValues.CONTAINS
+        )
+        content = ContentSequence()
+        if equivalent_meaning_of_value is not None:
+            if not isinstance(equivalent_meaning_of_value, str):
+                raise TypeError(
+                    'Argument "equivalent_meaning_of_value" must have type str.'
+                )
+            equivalent_meaning_of_value_item = TextContentItem(
+                name=codes.DCM.EquivalentMeaningOfValue,
+                value=equivalent_meaning_of_value,
+                relationship_type=RelationshipTypeValues.HAS_CONCEPT_MOD
+            )
+            content.append(equivalent_meaning_of_value_item)
+        if ecg_findings is not None:
+            if not isinstance(ecg_findings, (list, tuple, set)):
+                raise TypeError(
+                    'Argument "ecg_findings" must be a sequence.'
+                )
+            for ecg_finding in ecg_findings:
+                if not isinstance(ecg_finding, ECGFinding):
+                    raise TypeError(
+                        'Items of argument "ecg_finding" must have ' +
+                        'type ECGFinding.'
+                    )
+                content.extend(ecg_finding)
+        if len(content) > 0:
+            item.ContentSequence = content
+        super().__init__([item])
+
+
+class ECGQualitativeAnalysis(Template):
+    """:dcm:`TID 3717 <part16/sect_TID_3717.html>`
+    ECG Qualitative Analysis
+    """
+
+    def __init__(
+        self,
+        ecg_finding_text: Optional[str] = None,
+        ecg_finding_codes: Optional[Sequence[ECGFinding]] = None
```

Comment:
Minor, but perhaps there could just be one parameter `ecg_findings: Sequence[str | ECGFinding]` and then you could check the type internally to figure out what to do with them.

Reply:
Also, maybe "ecg" in the parameter names isn't needed, given the context?